openmmtools/mcmc.py

```diff
@@ -656,24 +662,37 @@ def apply(self, thermodynamic_state, sampler_state):
 
         # Perform the integration.
         for attempt_counter in range(self.n_restart_attempts + 1):
-            sampler_state.apply_to_context(context)
 
+            # If we reassign velocities, we can ignore the ones in sampler_state.
+            sampler_state.apply_to_context(context, ignore_velocities=self.reassign_velocities)
```

Comment:
This isn't a problem, but why does this function need to be able to ignore velocities at all? In asks the question "Why can't we ignore positions" (from a user perspective). Should this function instead just apply to context the whole sampler state, then the user can shuffle velocities like you do here in the next line?